lib/src/snapd_client.dart

```diff
@@ -187,6 +187,16 @@ class Snap with _$Snap {
   }) = _Snap;
 
   factory Snap.fromJson(Map<String, dynamic> json) => _$SnapFromJson(json);
+
+  static int _parseRevision(dynamic revision) {
+    if (revision is String) {
+      return int.tryParse(revision) ?? -1;
+    } else if (revision is int) {
+      return revision;
+    } else {
+      return -1;
+    }
+  }
```

Comment:
A locally installed snap revision can also be "x2" (which would correspond to -2 internally in snapd) - this wouldn't get parsed correctly here.
Concerning other APIs, such as the ratings, we'll have to make a distinction anyways, to exclude local snap revision we'd have to exclude negative revisions if we decide to parse them as integers. We could also keep them as strings and just check for a leading 'x' instead

Reply:
If we don't want to do type conversions further in we could also just turn the x to a *-1? I like converting things to their used types in the edges instead of doing type conversion in the middle, so that the code is more consistent.

Reply:
I can add a convenience getter here too, like ```dart
bool get isLocalRevision => revision.isNegative;
```
Then the code where we need to use it will be easier to read.

Reply:
> If we don't want to do type conversions further in we could also just turn the `x` to a `-` in the string before parsing it to an int? I like converting things to their used types in the edges instead of doing type conversion in the middle, so that the code is more consistent.

Yes, we need to do that before parsing to an int. Failure to parse a revision String, should raise an exception rather than defaulting to something like -1.



> I can add a convenience getter here too, like
```dart
bool get isLocalRevision => revision.isNegative;
```
> 
> Then the code where we need to use it will be easier to read.

So far we've tried to keep all the APIs of the client libraries as close to the ones they're exposing, and I think it'd be good to keep it that way. Wouldn't mind adding a convenience extension in the app center though.

Reply:
So we should expose the pure string here instead then is what you're saying? Or are you fine with int that throws an exception when it can't the string can't be parsed?

Reply:
Even though I don't expect snapd to send us wrong revision strings, I think parsing is the better solution, meaning: `"1234"`->`1234`, `"x3"`->`-3`, `"abcd"`->`Exception`